File: mobile-ios/Modules/Domain/Sources/Repositories/AdminRepository.swift
/// Mobil uygulamadaki admin ekranlarının veri kaynağı (yalnızca admin rolü).
public protocol AdminRepository: Sendable {
    func orders(status: OrderStatus?, language: String) async throws -> [Order]
    func order(id: String, language: String) async throws -> Order
    func updateOrderStatus(id: String, status: OrderStatus, note: String?, language: String)
        async throws -> Order

    func conversations() async throws -> [AdminConversation]
    func conversationMessages(id: String) async throws -> [Message]
    func reply(conversationID: String, content: String) async throws -> Message

    func lowStock(language: String) async throws -> [LowStockProduct]
}

File: mobile-android/app/src/main/java/com/erenlermarket/app/domain/repository/AdminRepository.kt
package com.erenlermarket.app.domain.repository

import com.erenlermarket.app.domain.model.AdminConversation
import com.erenlermarket.app.domain.model.LowStockProduct
import com.erenlermarket.app.domain.model.Message
import com.erenlermarket.app.domain.model.Order
import com.erenlermarket.app.domain.model.OrderStatus

/** Mobil uygulamadaki admin ekranlarının veri kaynağı. */
interface AdminRepository {
    suspend fun orders(status: OrderStatus?): List<Order>
    suspend fun order(id: String): Order
    suspend fun updateOrderStatus(id: String, status: OrderStatus, note: String?): Order

    suspend fun conversations(): List<AdminConversation>
    suspend fun conversationMessages(id: String): List<Message>
    suspend fun reply(conversationId: String, content: String): Message

    suspend fun lowStock(): List<LowStockProduct>
}
